androidgen/src/main/java/com/azure/autorest/android/template/AndroidClientMethodTemplate.java

```diff
@@ -0,0 +1,209 @@
+public class AndroidClientMethodTemplate extends ClientMethodTemplate {
+
+    private static ClientMethodTemplate _instance = new AndroidClientMethodTemplate();
+
+    protected AndroidClientMethodTemplate() {
+    }
+
+    public static ClientMethodTemplate getInstance() {
+        return _instance;
+    }
+
+    @Override
+    protected IType getContextType() {
+        return ClassType.AndroidContext;
+    }
+
+    @Override
+    protected void generatePagingSync(ClientMethod clientMethod, JavaType typeBlock, ProxyMethod restAPIMethod, JavaSettings settings) {
+    }
+
+    @Override
+    protected void generatePagingAsync(ClientMethod clientMethod, JavaType typeBlock, ProxyMethod restAPIMethod, JavaSettings settings) {
+    }
+
+    @Override
+    protected void generatePagedAsyncSinglePage(ClientMethod clientMethod, JavaType typeBlock, ProxyMethod restAPIMethod, JavaSettings settings) {
+        typeBlock.annotation("ServiceMethod(returns = ReturnType.SINGLE)");
+        writeMethod(typeBlock, clientMethod.getMethodVisibility(), clientMethod.getDeclaration(), function -> {
+            AddValidations(function, clientMethod.getRequiredNullableParameterExpressions(), clientMethod.getValidateExpressions(), settings);
+            AddOptionalAndConstantVariables(function, clientMethod, restAPIMethod.getParameters(), settings);
+            ApplyParameterTransformations(function, clientMethod, settings);
+            //ConvertClientTypesToWireTypes(function, clientMethod, restAPIMethod.getParameters(), clientMethod.getClientReference(), settings);
+
+            if (clientMethod.getMethodPageDetails().nonNullNextLink()) {
+                final String completeFutureVariableName = "completableFuture";
+                function.line(declareCompletableFuture(clientMethod, completeFutureVariableName));
+
+                final String callbackVariableName = "callbackVariable";
+                function.line(declareSinglePageCallback(restAPIMethod, completeFutureVariableName, callbackVariableName));
+
+                String serviceMethodCall = generateProxyMethodCall(clientMethod, restAPIMethod, settings, callbackVariableName);
+                function.line(serviceMethodCall);
+                function.methodReturn(completeFutureVariableName);
+            } else {
+                // REVISIT: Is there a use case for this?
+            }
+        });
+    }
+
+    private String declareSinglePageCallback(ProxyMethod restAPIMethod, String completeFutureVariableName, String callbackVariableName) {
+        ProxyMethodParameter callbackParam = restAPIMethod.getParameters().stream().filter(param -> param.getName().equals("callback")).findFirst().get();
+        IType callbackDataType = ((GenericType) callbackParam.getClientType()).getTypeArguments()[0];
+
+        StringBuilder callbackBuilder = new StringBuilder();
+        callbackBuilder.append(String.format("Callback<%1$s> %2$s = new Callback<%1$s>() {\n", callbackDataType, callbackVariableName));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Reply:
Thank you. I wasn't aware of that. I'll update this

Reply:
Happy to help :)